presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -1148,112 +1172,138 @@ public void testMaterializedViewOptimizationFullyMaterialized()
     @Test
     public void testMaterializedViewOptimizationNotMaterialized()
     {
-        String base = "orders_partitioned_not_materialized";
+        String table = "orders_partitioned_not_materialized";
         String view = "orders_partitioned_view_not_materialized";
 
         QueryRunner queryRunner = getQueryRunner();
         try {
             queryRunner.execute(format("CREATE TABLE %s WITH (partitioned_by = ARRAY['ds']) AS " +
                     "SELECT orderkey, orderpriority, '2020-01-01' as ds FROM orders WHERE orderkey < 1000 " +
                     "UNION ALL " +
-                    "SELECT orderkey, orderpriority, '2019-01-02' as ds FROM orders WHERE orderkey > 1000", base));
+                    "SELECT orderkey, orderpriority, '2019-01-02' as ds FROM orders WHERE orderkey > 1000", table));
 
             assertUpdate(format("CREATE MATERIALIZED VIEW %s WITH (partitioned_by = ARRAY['ds']) " +
-                    "AS SELECT orderkey, orderpriority, ds FROM %s", view, base));
+                    "AS SELECT orderkey, orderpriority, ds FROM %s", view, table));
             assertTrue(getQueryRunner().tableExists(getSession(), view));
 
-            String viewQuery = format("SELECT orderkey from %s where orderkey <  10000 ORDER BY orderkey", view);
-            String baseQuery = format("SELECT orderkey from %s where orderkey <  10000 ORDER BY orderkey", base);
+            String viewQuery = format("SELECT orderkey from %s where orderkey < 10000 ORDER BY orderkey", view);
+            String baseQuery = format("SELECT orderkey from %s where orderkey < 10000 ORDER BY orderkey", table);
 
             MaterializedResult viewTable = computeActual(viewQuery);
             MaterializedResult baseTable = computeActual(baseQuery);
             assertEquals(viewTable, baseTable);
+
+            assertPlan(getSession(), viewQuery, anyTree(
```

Comment:
Similarly lets check if the view scan node is not present.

Reply:
See [comment](https://github.com/prestodb/presto/pull/16561/files/004660be26968c6905417939af98e6c3c7d5fa23#r686204315)